sumologic/resource_sumologic_polling_source.go

```diff
@@ -75,19 +75,69 @@ func resourceSumologicPollingSource() *schema.Resource {
 		MaxItems: 1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice([]string{"S3BucketPathExpression", "CloudWatchPath"}, false),
+				},
 				"bucket_name": {
 					Type:     schema.TypeString,
-					Required: true,
+					Optional: true,
 					ForceNew: false,
 				},
 				"path_expression": {
 					Type:     schema.TypeString,
-					Required: true,
+					Optional: true,
+					ForceNew: false,
+				},
+				"limit_to_regions": {
+					Type:     schema.TypeList,
+					Optional: true,
+					ForceNew: false,
```

Comment:
I think these attributes such as `ForceNew` only need to be explicitly stated if they are `true`. Can you remove them and see if everything works?